tsp-output/@azure-tools/typespec-csharp/azoai/AzureOpenAI/AzureChatClient.cs

```diff
@@ -0,0 +1,107 @@
+﻿using AzureOpenAI.Models;
+using OpenAI;
+using OpenAI.Models;
+using System.ClientModel;
+using System.ClientModel.Primitives;
+
+namespace AzureOpenAI;
+
+internal class AzureChatClient : Chat
+{
+    private readonly string _apiVersion;
+    private readonly Uri _endpoint;
+
+    internal AzureChatClient(ClientPipeline pipeline, ApiKeyCredential credential, Uri endpoint, string apiVersion)
+        : base(pipeline, credential, endpoint)
+    {
+        _apiVersion = apiVersion;
+        _endpoint = endpoint;
+    }
+
+    public override async Task<ClientResult<CreateChatCompletionResponse>> CreateChatCompletionAsync(CreateChatCompletionRequest createChatCompletionRequest, CancellationToken cancellationToken = default)
+    {
+        Argument.AssertNotNull(createChatCompletionRequest, nameof(createChatCompletionRequest));
+
+        using BinaryContent content = createChatCompletionRequest.ToBinaryContent();
```

Comment:
Extension method on third-party type creates BinaryContent instance that writes content via internal Azure model wrapping third-party type.